lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1494,9 +1495,27 @@ def visit_join(self, join, asfrom=False, from_linter=None, **kwargs):
 
     def for_update_clause(self, select, **kw):
         if select._for_update_arg.read:
-            return " LOCK IN SHARE MODE"
+            tmp = " LOCK IN SHARE MODE"
         else:
-            return " FOR UPDATE"
+            tmp = " FOR UPDATE"
+
+        if select._for_update_arg.of:
```

Comment:
**Federico Caselli** (CaselIT) wrote:

Not sure if we should check here if the db support this?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1928

Reply:
I don't understand what you mean.
Mysql should always support this ?

Reply:
I meant version before the 8, from the issue it seems to be a new feature of version 8.
Also does mariadb support this? Currently they share this dialect

Reply:
Ah ok I understand now.
I think the comportment should be the same for all sgbd. For postgresql the version of the db is not checked. Do you agree ?
For mariadb I didn't know they share the same dialect. I will check maria db support it

Reply:
I guess that there is no check in pg since it support from before version 7.1 (the oldest one that has docs on the website).
But I agree that there is not a lot of consistency on what is checked and what it is not checked in general.

I'll leave the decision to mike.

Regarding mariadb, I guess it could be made more prominent in the documentation that the mysql dialect is shared by both

Reply:
You're right SKIP LOCKED and OF are not implemented for mariadb https://falseisnotnull.wordpress.com/2018/04/23/mysql-vs-mariadb-wait-nowait-skip-locked/
I change that

Reply:
let me know once you have updated it so that I can rerun the tests on gerrit

Reply:
I added a commit. I will check how to add a changelog

Reply:
I just checked some db driver requirements for postgres. 

psycopg2 - the minimum version supported is 7.4 (https://www.psycopg.org/docs/install.html#prerequisites)
pg8000 - minimum support is 9.4; this seems to be the active source - https://github.com/tlocke/pg8000
psycopg2cffi - doesn't seem to show any targeted versions
pypostgresql - can't find the minimum supported; this seems to be the active repo now https://github.com/python-postgres/fe via https://wiki.postgresql.org/wiki/Python
PyGresql - minimum PostgreSQL 9.0 (http://www.pygresql.org/)
jdbc - mimimum is 8.2 [see changelog v 42.0.0 https://jdbc.postgresql.org/documentation/changelog.html#version_42.0.0 )